web/src/click_handlers.js

```diff
@@ -924,4 +924,10 @@ export function initialize() {
     $(".settings-header.mobile .fa-chevron-left").on("click", () => {
         settings_panel_menu.mobile_deactivate_section();
     });
+
+    $("body").on("click", ".trigger-natural-click", (e) => {
+        // Jquery prevents default action on anchor for `trigger("click")`
+        // so we need to use click on element to trigger the default action.
+        e.currentTarget.click();
+    });
```

Comment:
Can you explain the motivation for this? I liked the previous model with explicit click handlers, both because it lets us set the `trigger` property (for logging) when adjusting the view and because that's the pattern we use on all of our other popovers, and this feels like a hacky workaround mechanism that, if it needed to exist, we'd have created quite some time ago.

Might be a good #frontend thread topic.

Reply:
I realized we use it commonly in our code:

```
    if ($(".dropdown.open, .dropup.open").length > 0 && $(e.target).attr("role") === "menuitem") {
        // on dropdown menu elements, force a click and prevent default.
        // this is because these links do not have an href and so don't force a
        // default action.
        e.target.click();
        return true;
```

Reply:
Although, the `role = menuitem` check is hard to guess what it does looking at the HTML code. So, we can migrate it to use `trigger-natural-click` class?

Let me start a frontend thread for this.

Reply:
Ok, changed to use click handlers. We still need `e.currentTarget.click();` to open the settings othewise we will have to use `window.location.hash = ...` which we don't like to.